### DISCUSSION TITLE
Blocks - Network association

### DISCUSSION THREAD

**Discussion by cata008** [Azure/ipam#173] [Ideas]
Hi @DCMattyG,

Hope you are doing well! 

I am back in business and have started using IPAM again. :) While using the tool, some ideas came to mind. 

1. While trying to create the network associations between my block and my Azure VNETs, I have received the error below because it seems there are conflicting IP ranges on the tenant. 
<img width="190" alt="image" src="https://github.com/Azure/ipam/assets/37331541/004c047d-9bf2-4780-a69f-32cc495f53c2">

My suggestion would be to return in the error message also which are the conflicting CIDRs. It would help a lot in finding which are the vnets causing it and cleaning them up or removing them from the association.

2. During the same network association process described above, I realized that hovering over vnets uses the same color as the already selected ones. Sometimes, if you go quickly over the list, you might omit selecting a vnet. Would it be possible to change one of the two colors? (already selected vnets or when hovering over new ones)
<img width="594" alt="image" src="https://github.com/Azure/ipam/assets/37331541/3a31eb8c-058f-4483-b2b1-620e90a73d8a">

3. Also, not sure if this is the case only for the network association window (don't believe so), but it would be nice to be able to drag the window on the screen.

4. Still the same topic - network associations. Is it possible to get somewhere, on the bottom left corner, a count of the total number of vnets and the ones that are already selected?

Cheers!

**Comment by DCMattyG** [comment]
Hey @cata008, allow me to address each of these items for you:

1. This is very doable, let me work on a good way to accomplish this.
2. The **hover** shade is actually *slightly* different than the **selected** shade, but I agree that this can be made more dramatic to avoid confusion.
3. Material UI, which is the framework I'm using here, does indeed have a simple ay to make the Dialog windows draggable, so I'll work on implementing that.
4. I'm sure this is doable, let me figure out the best way to make this look clean in the UI.

Thanks so much as always, this feedback absolutely helps to make Azure IPAM better for everyone!

**Reply by DCMattyG** [reply]
Hi @cata008, following up here with some updates...

1. I'm still thinking on this one. I agree that the output needs to be better, but I'm trying to figure out the best way to represent that. The *ideal* way would be with the full Resource ID's over the conflicting vNET's, but that would be a potentially very large error message. Just showing the overlapping CIDR's is a start, but there could also be multiple vNET's with the same CIDR, so the conflicts would still be hard to track down. Open to your thoughts here....
2. I'm looking into what this will take. I'm using a library to handle everything about these data grids and the colors are baked in. There is a way to create a custom CSS, but that does take a bit of effort. This is on my radar, so I'll keep working on this for you.
3. This is addressed in PR #181, all Dialog boxes are now draggable if you grab onto the top of them.
4. This is also addressed in PR #181. I ended up placing the count in the upper right after testing multiple different layouts. Hope this work for you!

In order to pull down the updated containers, simply restart your IPAM App Service:

![image](https://github.com/Azure/ipam/assets/12955732/d2ba398d-010e-4fc8-bce2-02485ac710ad)

Selected Network Count (Screenshot):

![image](https://github.com/Azure/ipam/assets/12955732/5a5e79fb-2667-438a-81e9-707e6aed0f0f)

Hope you enjoy all of the updates!

**Reply by cata008** [reply]
Hi @DCMattyG. Thanks for taking care of 3 & 4. Regarding 2, this is more of a cosmetic issue that is handy from my pov, but not super important. 
Now, about 1, I would start with the error message returning the conflicting CIDR(s). If there are mutiple duplicate CIDRs they will all have to be eventually addressed. I was thinking that it will help to have filtering on the Blocks page as well (and then filter based on the Name/Parent Space/CIDR). It would especially help with conflicting CIDRs even though sorting out and looking for the conflicting range is also an option.

**Reply by cata008** [reply]
@DCMattyG, on the same note, adding external networks on an existing block with vnets associated is not possible if there's a conflict.
That is expected, but there is no way to find out which is the conflicting range, apart from manually looking at all entries.

<img width="585" alt="image" src="https://github.com/Azure/ipam/assets/37331541/feb838eb-6abd-4468-9e3b-fa4ab3216413">
